ui/lib/pki/addon/components/pki-not-valid-after-form.ts

```diff
@@ -0,0 +1,78 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { HTMLElementEvent } from 'forms';
+import { format } from 'date-fns';
+
+/**
+ * <PkiNotValidAfterForm /> components are used to manage two mutually exclusive role options in the form.
+ */
+interface Args {
+  model: {
+    notAfter: string;
+    ttl: string | number;
+    set: (key: string, value: string | number) => void;
+  };
+}
+
+export default class RadioSelectTtlOrString extends Component<Args> {
+  @tracked groupValue = 'ttl';
+  @tracked cachedNotAfter: string;
+  @tracked cachedTtl: string | number;
+  @tracked formDate: string;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    const { model } = this.args;
+    this.cachedNotAfter = model.notAfter || '';
+    this.formDate = this.calculateFormDate(model.notAfter);
+    this.cachedTtl = model.ttl || '';
+    if (model.notAfter) {
+      this.groupValue = 'specificDate';
+    }
+  }
+
+  calculateFormDate(value: string) {
+    // API expects and returns full ISO string
+    // but the form input only accepts yyyy-MM-dd format
+    if (value) {
+      return format(new Date(value), 'yyyy-MM-dd');
+    }
+    return '';
+  }
+
+  @action onRadioButtonChange(selection: string) {
+    this.groupValue = selection;
+    // Clear the previous selection if they have clicked the other radio button.
+    if (selection === 'specificDate') {
+      this.args.model.set('ttl', '');
+      this.args.model.set('notAfter', this.cachedNotAfter);
```

Comment:
I'm never completely sure on this but I don't think `set` is needed and the value can be directly assigned.